portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/HeadlessDeliveryAPI/ExposeErcFromAssetLibraryForStructuredContents.testcase

```diff
@@ -40,6 +40,209 @@ definition {
+				assetLibraryId = "${depotId}",
+				filtervalue = "title%20eq%20%27WC%20Second%20WebContent%20Title%27");
+
+			HeadlessWebcontentAPI.assertProperNumberOfItems(
+				expectedTotalElement = "0",
+				responseToParse = "${response}");
+		}
+	}
+
+	@description = "Unable to get structured content by nonexistent assetLibraryId"
+	@priority = "4"
+	test CanReturnErrorWithGetStructuredContentInAssetLibraryByNonexistentAssetLibraryId {
+		property portal.acceptance = "true";
+
+		task ("When I make a GET request by non-existent assetLibraryID and a erc") {
+			var response = HeadlessWebcontentAPI.getStructuredContentsByErcInAssetLibrary(
+				assetLibraryId = "nonexistentId",
+				externalReferenceCode = "erc");
+		}
+
+		task ("Then I receive not found error response") {
+			TestUtils.assertPartialEquals(
+				actual = "${response}",
+				expected = "NOT_FOUND");
+		}
+
+		task ("And Then I receive error message in server console") {
+			AssertConsoleTextPresent(value1 = "Unable to get a valid asset library with ID nonexistentId");
+		}
+	}
+
+	@description = "Unable to get structured content by nonexistent erc"
+	@priority = "4"
+	test CanReturnErrorWithGetStructuredContentInAssetLibraryByNonexistentErc {
+		property portal.acceptance = "true";
+
+		task ("When I make a GET request by assetLibraryID and nonexistent erc") {
+			var depotId = JSONGroupAPI._getDepotIdByName(depotName = "Test Depot Name");
+
+			var response = HeadlessWebcontentAPI.getStructuredContentsByErcInAssetLibrary(
+				assetLibraryId = "${depotId}",
+				externalReferenceCode = "nonexistent-erc");
+		}
+
+		task ("Then I receive error message in response") {
+			TestUtils.assertPartialEquals(
+				actual = "${response}",
+				expected = "No JournalArticle exists with the key");
+		}
+	}
+
+	@description = "Structured content is updated by asset library id and erc"
+	@priority = "5"
+	test CanUpdateStructuredContentTitleInAssetLibraryByErc {
+		property portal.acceptance = "true";
+
+		task ("Given a structured content with a custom erc is created with a POST request in asset library") {
+			var depotId = JSONGroupAPI._getDepotIdByName(depotName = "Test Depot Name");
+			var ddmStructureId = WebContentStructures.getDdmStructureId(structureName = "content-structure");
+
+			var response = HeadlessWebcontentAPI.createStructuredContentInAssetLibrary(
+				assetLibraryId = "${depotId}",
+				data = "<p>My content</p>",
+				ddmStructureId = "${ddmStructureId}",
+				externalReferenceCode = "erc",
+				label = "Text",
+				name = "Text",
+				title = "WC WebContent Title");
+		}
+
+		task ("When I make a PUT request by assetLibraryId and erc") {
+			var response = HeadlessWebcontentAPI.updateStructuredContentByErcInAssetLibrary(
+				assetLibraryId = "${depotId}",
+				data = "<p>My content</p>",
+				ddmStructureId = "${ddmStructureId}",
+				externalReferenceCode = "erc",
+				label = "Text",
+				name = "Text",
```

Comment:
oh, I kinda see based on how we're using curl in poshi...it would require some other script to do this... :thinking: 

Ok, I get it better now after studying difference between PUT and PATCH methods. PUT will require all of these variables. I guess that's ok in order to conform with current _updateStructuredContentByErcInAssetLibrary macro...I'm not sure if I'm making this out to be a bigger deal than it is but I'm wondering, are all these other properties necessary outside of `assetLibraryID`, `externalReferenceCode`, and `title` if we're only just asserting an update on `title`?

Reply:
I think we need to include all these other properties during PUT /v1.0/asset-libraries/{assetLibraryId}/structured-contents/by-external-reference-code/{externalReferenceCode}. After tested, there will be an 400 BAD request error if we miss any other property. 
Actually, the `title` can be skipped if updating `data` part, since the test case is trying to update a title, so I just contain the `title` here.

Reply:
correct, for PUT request we need to pass all the required fields, in PATCH we can only pass a field we want to update